enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,498 @@
+  is well-tested and less likely to break than a Bash script.
+
+* As an OpenShift developer, I want to simplify the configure-ovs.sh script
+  by providing a better way to do advanced bridge configuration.
+
+### Goals
+
+Once this design is implemented, deployers of OpenShift will be able to
+explicitly configure br-ex to their exact specifications and will be able to
+modify it after deployment using standard networking tools.
+
+### Non-Goals
+
+Complete replacement of configure-ovs. This mechanism is intended for more
+advanced use cases, but it will require more complex configuration than
+basic deployments need. A possible followup would be to reimplement
+configure-ovs.sh in terms of this new mechanism so there is a simple path
+here too, but that is not required for the initial implementation.
+
+## Proposal
+
+This mechanism will be similar to the day-1 network configuration described
+in enhancements/network/baremetal-ipi-network-configuration.md. However,
+there will be some important differences, which will be discussed below.
+
+At a high level, we will provide an interface for the deployer to specify
+per-host interface configuration. Configuration for an OpenVSwitch bridge
+named br-ex will be mandatory. The format for the configuration will be
+NMState YAML, which will be provided to each node via Ignition. Around the
+same point in the boot process that configure-ovs would run, we will run
+NMState to apply the provided configuration. This will only be done on first
+boot because subsequent changes will be made using the Kubernetes-NMState
+operator.
+
+### Important differences from baremetal day-1
+
+Instead of pre-processing the NMState config into NetworkManager
+nmconnection files, we will simply write the NMState file directly to
+the host disk. As of 4.14 we have NMState available at the host level
+so it will no longer be necessary to run NMState in a container.
+
+Since this is intended to be a replacement for configure-ovs, it must
+work in a cross-platform manner. The baremetal feature is implemented in the
+baremetal-operator, which means non-baremetal platforms cannot use it.
+
+However, this is not a replacement for the baremetal feature (and any other
+platform-specific deploy-time network configuration tools). This is because
+minimal networking is required in order to pull Ignition, and since this
+configuration will be provided through Ignition it will still be necessary
+to have platform-specific methods of injecting network config before Ignition
+runs. However, the platform-specific part can be much simpler because it
+won't need to handle every possible network configuration, just enough to
+reach the Ignition endpoint. In some use cases, platform-specific configuration
+won't even be needed.
+
+### Workflow Description
+
+When the user is populating install-config they will provide the necessary
+configuration in the networking section (TODO: Is this the right place?).
```

Comment:
Do you mean a new networkConfig section in install-config? We can't use the existing one for this functionality because we don't have OVS in the ramdisk.

Reply:
You say we cannot use the existing one (referring to `networkConfig`, right?), yet the example below with br-ex is in the `networkConfig` block. I'm confused.

Reply:
Yeah, I reused the name but it's in a different section. I don't know if that's more or less confusing than coming up with a new name for a very similar feature.